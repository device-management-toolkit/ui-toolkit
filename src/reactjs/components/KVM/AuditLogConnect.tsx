/*********************************************************************
* Copyright (c) Intel Corporation 2019
* SPDX-License-Identifier: Apache-2.0
**********************************************************************/
import React from 'react';

const AuditLogConnect = ({deviceId,server}) =><button className="button" onClick={()=>window.location.href = `sampleAuditLog.htm?deviceId=${deviceId}&server=${server.split('/')[0]}`}>Fetch Audit Logs</button>

export default AuditLogConnect;