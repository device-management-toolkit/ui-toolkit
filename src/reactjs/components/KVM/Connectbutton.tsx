/*********************************************************************
 * Copyright (c) Intel Corporation 2019
 * SPDX-License-Identifier: Apache-2.0
 **********************************************************************/

import * as React from 'react'
require("./Connectbutton.scss")

export interface ConnectProps {
  kvmstate: number
  handleConnectClick: (e: any) => void
}

export class ConnectButton extends React.Component<ConnectProps, {}> {

  constructor(props: ConnectProps) {
    super(props);
  }


  render() {
    return (
      <button className="button" onClick={this.props.handleConnectClick}>
        {this.props.kvmstate == 1 ? 'Connecting KVM' : (this.props.kvmstate == 2 ? 'Disconnect KVM' : 'Connect KVM')}
      </button>
    )
  }
}