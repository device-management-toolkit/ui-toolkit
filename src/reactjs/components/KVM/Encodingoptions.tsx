/*********************************************************************
 * Copyright (c) Intel Corporation 2019
 * SPDX-License-Identifier: Apache-2.0
 **********************************************************************/

import * as React from 'react'
require("./Encodingoptions.scss");

export interface IEncodingOptions {
  changeEncoding: (encoding: number) => void
  getConnectState: () => number
}

export class EncodingOptions extends React.Component<IEncodingOptions, { value: number }> {

  constructor(props: IEncodingOptions) {
    super(props);
    this.state = { value: 1 };
    this.onEncodingChange = this.onEncodingChange.bind(this);
  }

  onEncodingChange(e) {
    //e.persist();
    this.setState({ value: e.target.value });
    this.props.changeEncoding(e.target.value);
  }

  render() {
    return (
      <span  className="encoding">
        <label >Encoding:</label>
        <select value={this.state.value} onChange={this.onEncodingChange} disabled={this.props.getConnectState() === 2}>
          <option value="1">RLE 8</option>
          <option value="2">RLE 16</option>
        </select>
      </span>
    )
  }
}