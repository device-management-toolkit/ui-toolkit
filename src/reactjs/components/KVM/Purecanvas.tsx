/*********************************************************************
 * Copyright (c) Intel Corporation 2019
 * SPDX-License-Identifier: Apache-2.0
 * Author : Ramu Bachala
 **********************************************************************/

import * as React from 'react';
require("./Purecanvas.scss");

export interface PureCanvasProps {
  contextRef: (ctx: CanvasRenderingContext2D) => void
  mouseDown: (event: React.MouseEvent) => void
  mouseUp: (event: React.MouseEvent) => void
  mouseMove: (event: React.MouseEvent) => void
  mouseWheel: (event: React.MouseEvent) => void
  canvasHeight: string
  canvasWidth: string
}

export class PureCanvas extends React.Component<PureCanvasProps, {}> {
  constructor(props: PureCanvasProps) {
    super(props);
  }
  
  shouldComponentUpdate() 
  { 
    return false; 
  }
  
  render() {
    
    let canvasAttributes : React.CanvasHTMLAttributes<HTMLCanvasElement> = {
      width:"400",
      height:"600",
      onContextMenu: (e) => {e.preventDefault(); return false;},
      onMouseDown: this.props.mouseDown,
      onMouseUp: this.props.mouseUp,
      onMouseMove: this.props.mouseMove,
      onWheel: this.props.mouseWheel
    }
    return (
        <canvas {...canvasAttributes} className="canvas" ref={c => c ? this.props.contextRef(c.getContext('2d')) : null}/> 
    )
  }
}

