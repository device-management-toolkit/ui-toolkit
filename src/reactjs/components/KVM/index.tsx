/*********************************************************************
* Copyright (c) Intel Corporation 2019
* SPDX-License-Identifier: Apache-2.0
**********************************************************************/
import React from 'react'
import ReactDOM from 'react-dom'
import { RemoteDesktop } from './UI'
import i18n from '../../../../i18n'
// Get browser language
i18n.changeLanguage(navigator.language).catch(() => console.info('error occured'))

const url = new URL(window.location.href)
const params = new URLSearchParams(url.search)

ReactDOM.render(<RemoteDesktop autoConnect={false} deviceId={params.get('deviceId')} mpsServer={params.get('mpsServer') + '/relay'} mouseDebounceTime={200} canvasHeight={'100%'} canvasWidth={'100%'} />, document.querySelector('#kvm'))
