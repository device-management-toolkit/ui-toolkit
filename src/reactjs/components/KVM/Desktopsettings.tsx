/*********************************************************************
 * Copyright (c) Intel Corporation 2019
 * SPDX-License-Identifier: Apache-2.0
 **********************************************************************/

import * as React from 'react'
import { EncodingOptions } from './Encodingoptions'

export interface IDesktopSettings {
  changeDesktopSettings: (settings: any) => void
  getConnectState: () => number
}

export class DesktopSettings extends React.Component<IDesktopSettings> {
  desktopsettings = {
    encoding: 1
  };

  constructor(props: IDesktopSettings) {
    super(props);
    this.changeEncoding = this.changeEncoding.bind(this);
  }

  changeEncoding(encoding: number) {
    this.desktopsettings.encoding = encoding;
    this.props.changeDesktopSettings(this.desktopsettings);
  }

  render() {
    return (
      <EncodingOptions changeEncoding={this.changeEncoding} getConnectState={this.props.getConnectState}/>
    )
  }
}