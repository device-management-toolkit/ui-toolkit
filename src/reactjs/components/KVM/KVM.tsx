/*********************************************************************
 * Copyright (c) Intel Corporation 2019
 * SPDX-License-Identifier: Apache-2.0
 * Author : Ramu Bachala
 **********************************************************************/

import * as React from 'react'
import { PureCanvas } from './Purecanvas';
import { componentFromStream, createEventHandler } from 'recompose';
import { tap, map, debounceTime, catchError, merge, startWith, throttleTime } from 'rxjs/operators'
import { combineLatest } from 'rxjs'

const KVM = componentFromStream((prop$) => {
  const { handler: mouseDown, stream: mouseDown$ } = createEventHandler();
  const { handler: mouseUp, stream: mouseUp$ } = createEventHandler();
  const { handler: mouseMove, stream: mouseMove$ } = createEventHandler();
  const { handler: mouseWheel, stream: mouseWheel$ } = createEventHandler();

  const events1$ = mouseMove$.pipe(startWith(new MouseEvent('mousemove')), throttleTime(200)) // dont send mousemove more than once in 200 ms
  const events2$ = mouseUp$.pipe(startWith(new MouseEvent('mouseup')))
  const events3$ = mouseDown$.pipe(startWith(new MouseEvent('mousedown')))

  var getCanvas$ = combineLatest(prop$, events1$, events2$, events3$).pipe(
    map(
      ([{ saveContext,canvasHeight,canvasWidth,mousemove, mousedown, mouseup }, event]: any) => { // what you get is a 2 element array with this.props and mouse event
        switch (event.type) {
          case 'mousemove':
            mousemove(event)
            break;
          case 'mousedown':
            mousedown(event)
            break;
          case 'mouseup':
            mouseup(event)
            break;
          default:
            break;
        }
        return <PureCanvas contextRef={saveContext} canvasHeight={canvasHeight} canvasWidth={canvasWidth}
          mouseDown={mouseDown}
          mouseUp={mouseUp}
          mouseMove={mouseMove}
          mouseWheel={mouseWheel} />
      }
    ),
    catchError((e) => { console.log(e); return null; })
  )

  return getCanvas$
});

export { KVM }