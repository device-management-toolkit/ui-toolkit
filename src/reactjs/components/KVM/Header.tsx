/*********************************************************************
 * Copyright (c) Intel Corporation 2019
 * SPDX-License-Identifier: Apache-2.0
 **********************************************************************/

import * as React from 'react'
import { ConnectButton } from './Connectbutton'
import { DesktopSettings } from './Desktopsettings'
import AuditLogConnect from './AuditLogConnect';
require("./Header.scss");

export interface IHeaderProps {
  kvmstate: number,
  deviceId: string,
  server: string,
  handleConnectClick: (e: any) => void
  changeDesktopSettings: (settings: any) => void
  getConnectState: () => number
}

export class Header extends React.Component<IHeaderProps> {

  constructor(props: IHeaderProps) {
    super(props);
  }

  render() {
    return (
      <div className="header">
        <AuditLogConnect deviceId={this.props.deviceId} server={this.props.server}/>
        <ConnectButton handleConnectClick={this.props.handleConnectClick} kvmstate={this.props.kvmstate} />
        <DesktopSettings changeDesktopSettings={this.props.changeDesktopSettings} getConnectState={this.props.getConnectState}/>
      </div >
    )
  }
}