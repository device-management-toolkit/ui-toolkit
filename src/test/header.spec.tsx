/*********************************************************************
* Copyright (c) Intel Corporation 2019
* SPDX-License-Identifier: Apache-2.0
**********************************************************************/

import * as React from 'react';
import {IHeaderProps, Header} from '../reactjs/components/KVM/Header'
import {shallow} from 'enzyme'

describe('Testing Header',()=>{
  it('Test render() in Header',()=>{
    // Initialization of IHeaderProps
    let headerprops : IHeaderProps= {
        kvmstate: 1, 
        deviceId:'abcd-11',
        server:'9300',
        handleConnectClick: (testFunc2),
        changeDesktopSettings: (testFunc3),
        getConnectState: (testFunc1)     
    };

    const he = shallow(<Header {...headerprops}  />);
    let myInstance = he.instance() as Header;
    
    // Output    
    expect(he.find('ConnectButton').prop('handleConnectClick')).toBe(testFunc2);
    expect(he.find('ConnectButton').prop('kvmstate')).toBe(1);
    expect(he.find('DesktopSettings').prop('changeDesktopSettings')).toBe(testFunc3);
    expect(he.find('DesktopSettings').prop('getConnectState')).toBe(testFunc1);
    var ret = expect(he).toMatchSnapshot();
    console.log(he.debug());
    console.log(he.props());
  });
});

function testFunc1(): number{
  return 1;
}

class TestClass
{
  encoding: number;
}

var value1 = 0;

function testFunc2(v: TestClass): void{
  value1 = v.encoding;
}

function testFunc3(v: TestClass): void{
  value1 = v.encoding;  
}
  


