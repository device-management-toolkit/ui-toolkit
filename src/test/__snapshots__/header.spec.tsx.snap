// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Header Test render() in Header 1`] = `ShallowWrapper {}`;
